Mathlib/RingTheory/Polynomial/Quotient.lean

```diff
@@ -54,183 +64,228 @@ open Polynomial
 
 variable {R : Type _} [CommRing R]
 
-theorem quotient_map_c_eq_zero {I : Ideal R} :
+theorem quotient_map_C_eq_zero {I : Ideal R} :
     ∀ a ∈ I, ((Quotient.mk (map (C : R →+* R[X]) I : Ideal R[X])).comp C) a = 0 := by
   intro a ha
-  rw [RingHom.comp_apply, quotient.eq_zero_iff_mem]
+  rw [RingHom.comp_apply, Quotient.eq_zero_iff_mem]
   exact mem_map_of_mem _ ha
-#align ideal.quotient_map_C_eq_zero Ideal.quotient_map_c_eq_zero
+#align ideal.quotient_map_C_eq_zero Ideal.quotient_map_C_eq_zero
 
-theorem eval₂_c_mk_eq_zero {I : Ideal R} :
+theorem eval₂_C_mk_eq_zero {I : Ideal R} :
     ∀ f ∈ (map (C : R →+* R[X]) I : Ideal R[X]), eval₂RingHom (C.comp (Quotient.mk I)) X f = 0 := by
   intro a ha
   rw [← sum_monomial_eq a]
   dsimp
   rw [eval₂_sum]
-  refine' Finset.sum_eq_zero fun n hn => _
+  refine' Finset.sum_eq_zero fun n _ => _
   dsimp
-  rw [eval₂_monomial (C.comp (Quotient.mk' I)) X]
+  rw [eval₂_monomial (C.comp (Quotient.mk I)) X]
   refine' mul_eq_zero_of_left (Polynomial.ext fun m => _) (X ^ n)
   erw [coeff_C]
   by_cases h : m = 0
-  · simpa [h] using quotient.eq_zero_iff_mem.2 ((mem_map_C_iff.1 ha) n)
+  · simpa [h] using Quotient.eq_zero_iff_mem.2 ((mem_map_C_iff.1 ha) n)
   · simp [h]
-#align ideal.eval₂_C_mk_eq_zero Ideal.eval₂_c_mk_eq_zero
+#align ideal.eval₂_C_mk_eq_zero Ideal.eval₂_C_mk_eq_zero
+
+lemma polynomialQuotientEquivQuotientPolynomial_eval₂RingHom_leftInverse (I : Ideal R) :
```

Comment:
Could you please leave a `-- Porting note: ` explaining why you pulled out these proofs into standalone lemmas?

Reply:
I restored this one, but `quotientEquivQuotientMvPolynomial` is still too slow; added a porting note for that.